vllm/v1/core/scheduler.py

```diff
@@ -329,6 +362,22 @@ def schedule(self) -> "SchedulerOutput":
                 self.kv_cache_manager.get_num_common_prefix_blocks(
                     any_request, len(self.running)))
 
+        # Prepare the guided decoding bitmask for this batch.
+        grammar_bitmask = None
+        if guided_decoding_request_ids:
+            # Fill the bitmask using the index of each request equal to its
+            # position in the batch. Resize the bitmask down to the size of
+            # the batch.
+            grammar_bitmask = self.guided_decoding_manager.grammar_bitmask
+            assert grammar_bitmask is not None
+            for req_id, batch_index in guided_decoding_request_ids.items():
+                request = self.requests[req_id]
+                assert request.grammar is not None
+                if not request.grammar.matcher.is_terminated():
+                    request.grammar.fill_bitmask(grammar_bitmask, batch_index)
+            if len(self.running) < grammar_bitmask.shape[0]:
+                grammar_bitmask = grammar_bitmask[:len(self.running)]
```

Comment:
This is for follow-up. I don't think we need to block on it.

When this was originally written, I thought we'd be re-using this bitmask as-is in the GPU worker. However, the GPU worker maintains its own batch data structures that are not in the same order as the scheduler. We end up having to reconstruct the bitmask there, so there's no good reason to have one that has empty unused rows in it. We should just create one that has a size equal to the number of guided requests to save some space.